### ISSUE TITLE
Comparing two varbits for equality

### ISSUE THREAD

**Issue by mbudiu-vmw** [p4lang/behavioral-model#536]
The P4-16 spec is adding an operation that can compare two varbit values for equality.
Using equality testing for varbits in json seems to be accepted by bmv2, but does not seem to produce the right result.

**Comment by antoninbas** [comment]
Is it possible to have the JSON file for the program?

**Comment by mbudiu-vmw** [comment]
[equality-varbit.zip](https://github.com/p4lang/behavioral-model/files/1662242/equality-varbit.zip)

**Comment by mbudiu-vmw** [comment]
This is the gist of the program: 
```
header H {
    bit<8>     s;
    varbit<32> v;
}

control ingress(...) {
    apply {
        stdmeta.egress_spec = 0;
        H h = hdr.h;

        if (hdr.h.v == h.v) {
            stdmeta.egress_spec = 1;
        }
    }
}
```

**Comment by lomosewen** [comment]
Have you checked if the varbit field was extracted correctly? I also have some problems with varbit stuff. What I found is that my varbit field were all set to 0 after parsing. At the same time, the fixed-length fields all work fine.

**Comment by antoninbas** [comment]
@mbudiu-vmw this has the same root cause as #538 which means it is fixed by #539. I've confirmed it by running your test before and after the fix.
